### PR TITLE
Set Sparse compression to save_compressed

### PR DESCRIPTION
**Pull request by rahul-tuli**
PR Description: Before this PR, sparse compression was not being utilized since it was explicitly set to False in the codebase. This PR enables sparse compression, allowing us to take advantage of its benefits for reduced memory on disk

### CODE REVIEW

File: src/llmcompressor/transformers/sparsification/compressed_tensors_utils.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def save_pretrained_wrapper(
                     "skip_compression_stats=True"
                 )
                 sparsity_config = SparsityConfigMetadata.from_pretrained(
-                    model, state_dict=state_dict, compress=False
+                    model, state_dict=state_dict, compress=save_compressed
                 )
```
Comment (mgoin) [inline src/llmcompressor/transformers/sparsification/compressed_tensors_utils.py:99]: Do you think we should change `save_compressed: bool = True` to False in order to preserve previous behavior?

**Review by mgoin — CHANGES_REQUESTED** [review]

**Review by mgoin — APPROVED** [review]
Per offline discussion, we will allow this automatic behavior
